### PR TITLE
Remove some StringBuilderCache usage from Corelib

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs [hunk 1]
```diff
@@ -246,15 +246,12 @@ internal void BackOne()
...
+            internal char NextChar()
             {
-                get
-                {
-                    int pos = ++_pos;
-                    return (uint)pos < (uint)_value.Length ?
-                        _value[pos] :
-                        (char)0;
-                }
+                int pos = ++_pos;
```
Comment (danmoseley) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs:251]: For some reason I thought we conventionally do post fix in such cases

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs:251]: That would change the meaning. Can you clarify?

Comment (danmoseley) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs:251]: Reading too fast 😀

File: src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs [hunk 2]
```diff
@@ -1324,18 +1321,18 @@ private static bool TryParseByFormat(ReadOnlySpan<char> input, ReadOnlySpan<char
 
                     case '\'':
                     case '\"':
-                        StringBuilder enquotedString = StringBuilderCache.Acquire();
-                        if (!DateTimeParse.TryParseQuoteString(format, i, enquotedString, out tokenLen))
+                        var enquotedString = new ValueStringBuilder(32);
```
Comment (danmoseley) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs:1324]: would it make sense to be slightly less conservative? 64?

File: src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs [hunk 3]
```diff
@@ -1440,11 +1437,12 @@ private static bool ParseExactDigits(ref TimeSpanTokenizer tokenizer, int minDig
...
 
-        private static bool ParseExactLiteral(ref TimeSpanTokenizer tokenizer, StringBuilder enquotedString)
+        private static bool ParseExactLiteral(ref TimeSpanTokenizer tokenizer, ref ValueStringBuilder enquotedString)
         {
-            for (int i = 0; i < enquotedString.Length; i++)
+            ReadOnlySpan<char> span = enquotedString.AsSpan();
+            for (int i = 0; i < span.Length; i++)
             {
-                if (enquotedString[i] != tokenizer.NextChar)
+                if (span[i] != tokenizer.NextChar())
```
Comment (danmoseley) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs:1445]: what is the value of the Span here -- is indexing into it faster than indexing into a string?

**Review by danmoseley — APPROVED** [review]

**Comment by danmoseley** [comment]
BTW, I noticed the comment higher up StreamWriter about https://github.com/dotnet/runtime/issues/8890

The docs I've seen say that it's not guaranteed that stackallocs are zero inited - although the issue above implies we always do. Should we document that we do without SkipLocalsInit, and don't with SkipLocalsInit? Otherwise it's hard to reason about code.

File: src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs (hunk 3 above)
Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs:1445]: It wasn't a string. It was a StringBuilder, and yes, indexing into a span is way faster than indexing into a StringBuilder.

File: src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs (hunk 2 above)
Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs:1324]: sure

File: src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs (hunk 3 above)
Comment (danmoseley) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs:1445]: I of all people should know that: https://github.com/dotnet/runtime/issues/24395. Perhaps I should open an analyzers suggestion.

**Pull request merged**